server/src/main/java/org/apache/uniffle/server/merge/DefaultMergeEventHandler.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.server.merge;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import com.google.common.collect.Queues;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.uniffle.common.util.ThreadUtils;
+import org.apache.uniffle.server.ShuffleServerConf;
+import org.apache.uniffle.server.ShuffleServerMetrics;
+
+import static org.apache.uniffle.server.ShuffleServerConf.SERVER_MERGE_THREAD_ALIVE;
+import static org.apache.uniffle.server.ShuffleServerConf.SERVER_MERGE_THREAD_POOL_QUEUE_SIZE;
+import static org.apache.uniffle.server.ShuffleServerConf.SERVER_MERGE_THREAD_POOL_SIZE;
+
+public class DefaultMergeEventHandler implements MergeEventHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultMergeEventHandler.class);
+
+  private Executor threadPoolExecutor;
+  protected final BlockingQueue<MergeEvent> queue = Queues.newLinkedBlockingQueue();
+  private Consumer<MergeEvent> eventConsumer;
+  private volatile boolean stopped = false;
+
+  public DefaultMergeEventHandler(
+      ShuffleServerConf serverConf, Consumer<MergeEvent> eventConsumer) {
+    this.eventConsumer = eventConsumer;
+    int poolSize = serverConf.get(SERVER_MERGE_THREAD_POOL_SIZE);
+    int queueSize = serverConf.get(SERVER_MERGE_THREAD_POOL_QUEUE_SIZE);
+    int keepAliveTime = serverConf.get(SERVER_MERGE_THREAD_ALIVE);
+    BlockingQueue<Runnable> waitQueue = Queues.newLinkedBlockingQueue(queueSize);
+    threadPoolExecutor =
```

Comment:
We should better set exceptionHandler for thread pool.

Reply:
@jerqi I catch exception in [here](https://github.com/apache/incubator-uniffle/blob/dc2d27a9cbfbfed51b4416c900ef31f1c253c1df/server/src/main/java/org/apache/uniffle/server/merge/PartitionEntity.java#L216). 
For now, If merge fail, the partition will set to INTERNAL_ERROR state. The reduce will get error. However, we will also add retry in the future